src/consumer.coffee

```diff
@@ -1,7 +1,52 @@
+crypto            = require 'crypto'
+URL         = require 'url-parse'
+HMAC_SHA1         = require './hmac-sha1'
+errors            = require './errors'
+extensions        = require './extensions'
+
 class Consumer
-  constructor: () ->
-    console.error('Consumer Not Implemented')
-    false
+  constructor: (consumer_key, consumer_secret) ->
+    if typeof consumer_key is 'undefined' or consumer_key is null
+      throw new errors.ConsumerError 'Must specify consumer_key'
+
+    if typeof consumer_secret is 'undefined' or consumer_secret is null
+      throw new errors.ConsumerError 'Must specify consumer_secret'
+
+    @consumer_key     = consumer_key
+    @consumer_secret  = consumer_secret
+    @signer           = new HMAC_SHA1()
+    @body             = {}
+
+  sign_request: (url, method = 'GET', body = {}) =>
```

Comment:
Maybe i've been generalizing this a bit too much. 

The reason i've added this in is because as a tool provider using the ContentItemMessage protocol I need to send data back to the tool consumer that is signed using the key/secret. I could refactor this and put it into the Provider under something regarding signing the ContentItemMessage that needs to be sent back to the tool consumer. That being said it wouldn't necessarily have anything lti-specific in its implementation, but as a tool provider if I am using this library already I wouldn't really want to get another one to sign the message.